epub3-to-daisy202/src/main/resources/xml/xslt/augment-smil.xsl

```diff
@@ -0,0 +1,114 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0"
+                xmlns:html="http://www.w3.org/1999/xhtml"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:pf="http://www.daisy.org/ns/pipeline/functions"
+                xmlns="" xpath-default-namespace=""
+                exclude-result-prefixes="#all">
+    
+    <xsl:import href="http://www.daisy.org/pipeline/modules/file-utils/library.xsl"/>
+    
+    <!--
+        the smil
+    -->
+    <xsl:variable name="smil" select="collection()[1]/*"/>
+    <!--
+        the corresponding content document(s) in spine order
+    -->
+    <xsl:variable name="html" select="collection()[position()&gt;1]/*"/>
+    
+    <xsl:variable name="smil-base-uri" select="base-uri($smil)"/>
+    
+    <xsl:key name="absolute-id" match="*[@id]" use="concat(base-uri(root()/*),'#',@id)"/>
+    
+    <xsl:variable name="referenced-html-elements" as="element()*">
+        <xsl:for-each select="//text">
+            <xsl:variable name="absolute-src" select="resolve-uri(@src,$smil-base-uri)"/>
```

Comment:
You're right. I thought about that but indeed ignored it because I thought it would be rare. But maybe it is not so rare after all. You can for example specify a relative xml:base (something I first thought was not allowed but after some research found that it is).

A problem with resolving URIs against the base URI of the element in question and not the root element, is that the Saxon/Calabash update has introduced some bugs so we have to be careful. For more info see https://github.com/daisy/pipeline-modules-common/commit/24a9e9107715e583e05a48d1d182ab5254a8f64f, and I will also add some unit tests for this problem later.

Reply:
See https://github.com/daisy/pipeline-modules-common/commit/8c987d91bd23a99c6b39b048a96b6a6931e9b921